### ISSUE TITLE
CanvasCaptureMediaStreamTrack as media option

### ISSUE THREAD

**Issue by km147** [andyet/SimpleWebRTC#516]
Hi,

I was wondering if its possible to send a CanvasCaptureMediaStreamTrack object as a media constraint?

I have tried the following:

`var videotracks = canvas.captureStream().getTracks();
var videostream = videotracks[0];

	var webrtc = new SimpleWebRTC({
	   localVideoEl: '',
	   remoteVideosEl: '',
	   autoRequestMedia: false,
	   receiveMedia: {
		offerToReceiveAudio: 0,
		offerToReceiveVideo: 0
	   },
	    media: { video: {optional: [{sourceId: videostream.id}]} }
	});

This results in the peer stream being undefined for receiving clients. Can you please point me in the right direction?

**Comment by fippo** [comment]
canvas capture streams don't work via getUserMedia (which is what the media settings are for).

Try setting autoRequestMedia to false, then push the captureStream result to webrtc.webrtc.localStreams and re-evaluating testReadiness

**Comment by km147** [comment]
Ok got it, many thanks

**Comment by russelldc** [comment]
@km147 How did you solve this issue? I've tried pushing the cpatured stream to webrtc.webrtc.localStreams, but nothing appears to happen.

**Comment by wardhanster** [comment]
@russelldc Did you managed to get it working ??

**Comment by russelldc** [comment]
@wardhanster Nope

**Comment by wardhanster** [comment]
@russelldc that's a bummer, did you tried something else, maybe another platform to have that working?

**Comment by russelldc** [comment]
@wardhanster No, sorry! I ended up not needing to stream a canvas at all after re-assessing my requirements.
